lms/views/lti/basic_launch.py

```diff
@@ -294,8 +282,35 @@ def _show_document(
             extra=assignment_extra,
         )
 
+        # An assignment has been configured in the LMS as "gradable" if it has
+        # the `lis_outcome_service_url` param
+        gradable = bool(self.context.lti_params.get("lis_outcome_service_url"))
```

Comment:
I think this conditional could be:

`def is_gradable` in LTITResource

I can't point to the specific use case but if we split this view in multiple ones or we wanted to use this in deep-linking (which also uses LTIResoruce, it's also an LTI launch of a different type) we could re-use it.

It just feels like the right place IMO.

Also, not sure about naming but we might want to have both:

`is_assignment_gradable` (does the assignment it self support grading)
`is_user_gradable` (can the user himself have a grade)


I think having those two would help using 

 `_add_canvas_speedgrader_settings` here. 
 Currently that  happens all the time with ` self.context.js_config.add_document_url(document_url)` and  `_add_canvas_speedgrader_settings` has a very similar, duplicated conditional soup.

Reply:
I've pulled the speed grader stuff in to, but I'd like to leave off moving things back out of the view again for is gradable until there's some clear demand for it.